Source/Dafny/Resolver.cs

```diff
@@ -436,6 +439,14 @@ enum ValuetypeVariety {
 
       rewriters.Add(new InductionRewriter(reporter));
 
+      if (DafnyOptions.O.Plugins.Count() >= 1) {
```

Comment:
Use `Length` over `Count()`. The latter enumerates all the items to compute the count, unless there are some optimizations underneath (there are some in LINQ that check for certain types).
On the other hand, since it's only a guard for the `foreach` loop on the next line, you can skip this altogether. If there are no items, no iteration will happen.